### PR TITLE
Carry #27834 — Do not require `.git` in the build context

### PR DESCRIPTION
**Pull request by vdemeester**
Carry #27834 and hopefully fixes builds :angel:.

- Add `.git` to `.dockerignore`
- export DOCKER_GITCOMMIT in the Makefile
- prioritize DOCKER_GITCOMMIT against the `git` command in
  `./hack/make.sh`
- Also add `integration-cli/bundles` to gitignore (it's generated when
  using integration-cli shell)

Note that, on my laptop for example, I'm sending `30M` of context with this, and `150M` without :scream_cat:, so I really think this can be a huge win :angel: 

/cc @tianon @jhowardmsft @icecrime @thaJeztah @cpuguy83 

:frog:

### CODE REVIEW

**Comment by vdemeester** [comment]
Hum, I really don't know why we're not using `make …` on the CI and instead manual `docker run …`, really :sweat:

File: .dockerignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 bundles
+integration-cli/bundles
```
Comment (AkihiroSuda) [inline .dockerignore]: Just a thought: I think this directory should be removed and unified to the top-level bundle directory, but probably it should be another PR

**Comment by tianon** [comment]
> Hum, I really don't know why we're not using `make …` on the CI and instead manual `docker run …`, really :sweat:

IIRC, it was a security concern -- if the CI were using the `Makefile` directly, I could send a PR which simply updates the `Makefile` with some malicious actions and they'll be run directly on the host.  By forcing the CI to use `docker build ... && docker run ...` directly, we insulate the CI from that vector.

**Comment by vdemeester** [comment]
@tianon hum I see :sweat:

**Comment by vdemeester** [comment]
Waiting for the 1.13 release to update CI… I wouldn't want to break anything right now :angel:

**Comment by icecrime** [comment]
@vdemeester Is it ok to close this one until 1.13.0 is out of the door?

**Comment by vdemeester** [comment]
Right, I'll reopen a little bit later 👼
